### PR TITLE
New examples

### PR DESCRIPTION
**Pull request by paulstapor**
Very most of the new functionality is implemented into the rosenbrock example, but some things can also be found in the petab example...

### CODE REVIEW

**Comment by paulstapor** [comment]
Now, I think I really double checked all the stuff or at least 95% of it...
I hope, I found most of the obvious errors...

File: pypesto/visualize/optimizer_history.py [hunk 1]
```diff
@@ -308,6 +308,9 @@ def get_vals(vals, scale_y, offset_y, start_indices):
 
     vals: list
         list of numpy arrays of size 2 x len(start_indices)
+
+    offset_y:
+        offset for the y-axis, as this is supposed to be in log10-scale
```
Comment (yannikschaelte) [inline pypesto/visualize/optimizer_history.py]:
```suggestion
        offset for the y-axis, if this is supposed to be in log10-scale
```
?

File: pypesto/visualize/parameters.py [hunk 2]
```diff
@@ -49,6 +49,10 @@ def parameters(results, ax=None, free_indices_only=True, lb=None, ub=None,
         Flag indicating whether alpha for large clusters should be reduced to
         avoid overplotting (default: True)
 
+    start_indices: list or int
+        list of integers specifying the multistart to be plotted or
```
Comment (yannikschaelte) [inline pypesto/visualize/parameters.py]:
```suggestion
        list of integers specifying the multistarts to be plotted or
```

File: pypesto/visualize/parameters.py [hunk 3]
```diff
@@ -198,6 +218,10 @@ def handle_inputs(result, free_indices_only, lb=None, ub=None):
         If not None, override result.problem.lb, problem.problem.ub.
         Dimension either result.problem.dim or result.problem.dim_full.
 
+    start_indices: list or int
+        list of integers specifying the multistart to be plotted or
```
Comment (yannikschaelte) [inline pypesto/visualize/parameters.py]:
```suggestion
        list of integers specifying the multistarts to be plotted or
```

File: pypesto/visualize/profiles.py [hunk 4]
```diff
@@ -6,8 +6,8 @@
 from .misc import process_result_list
 
 
-def profiles(results, fig=None, profile_indices=None, size=(18.5, 6.5),
-             reference=None, colors=None, legends=None):
+def profiles(results, ax=None, profile_indices=None, size=(18.5, 6.5),
+             reference=None, colors=None, legends=None, profile_list=0):
```
Comment (yannikschaelte) [inline pypesto/visualize/profiles.py]: ``profile_list`` sounds like one would pass a list. ``profile_list_ix/index``?

File: pypesto/visualize/profiles.py [hunk 5]
```diff
@@ -248,10 +252,14 @@ def handle_reference_points(ref, ax, fvals):
...
 
+            # create legend for reference points
+            if i_ref.legend is not None:
+                i_ax.legend()
+
     return ax
 
 
-def handle_inputs(result, profile_indices):
+def handle_inputs(result, profile_indices, profile_list=0):
```
Comment (yannikschaelte) [inline pypesto/visualize/profiles.py]: see above

File: pypesto/visualize/profiles.py [hunk 6]
```diff
@@ -276,9 +287,11 @@ def handle_inputs(result, profile_indices):
     fvals = []
     for i_par in range(0, len(result.profile_result.list[0])):
         if i_par in profile_indices:
-            tmp = np.array(
-                [result.profile_result.list[0][i_par].x_path[i_par, :],
-                 result.profile_result.list[0][i_par].ratio_path[:]])
+            tmp = np.array([result.profile_result.list[profile_list][
```
Comment (yannikschaelte) [inline pypesto/visualize/profiles.py]: sorry, but this code looks sooo autoformatted :)

File: pypesto/visualize/waterfall.py [hunk 7]
```diff
@@ -208,6 +208,9 @@ def get_fvals(result, scale_y, offset_y, start_indices):
 
     fvals: ndarray
         function values
+
+    offset_y:
+        offset for the y-axis, if this is supposed to be in log10-scale
```
Comment (yannikschaelte) [inline pypesto/visualize/waterfall.py]:
```suggestion
        offset for the y-axis, if it is supposed to be in log10-scale
```

File: pypesto/visualize/waterfall.py [hunk 8]
```diff
@@ -265,6 +268,9 @@ def handle_options(ax, max_len_fvals, ref, y_limits):
     y_limits: float or ndarray, optional
         maximum value to be plotted on the y-axis, or y-limits
 
+    offset_y:
+        offset for the y-axis, if this is supposed to be in log10-scale
```
Comment (yannikschaelte) [inline pypesto/visualize/waterfall.py]:
```suggestion
        offset for the y-axis, if it is supposed to be in log10-scale
```

File: pypesto/visualize/profiles.py (hunk 4 above)
Comment (paulstapor) [reply pypesto/visualize/profiles.py]: puh... right, that's not finished, I decided to give up on this... 😅 
The idea is to have a list of axis handles for a profiling plot, which on can pass to another call of the profiling plot... There are arbitrarily many problems with that, however... E.g. what, if you want to plot for another number of profiles than you have entries in the handle list? How to make sure that these profiles, which are plotted in the same axes, really belong to the same parameter? Etc. etc...

File: pypesto/visualize/profiles.py (hunk 6 above)
Comment (paulstapor) [reply pypesto/visualize/profiles.py]: yes, arbitrarily ugly...
will change that...

File: pypesto/visualize/profiles.py (hunk 4 above)
Comment (yannikschaelte) [reply pypesto/visualize/profiles.py]: if it's not used yet and maybe too difficult to implement, I'd just remove it?

Comment (paulstapor) [reply pypesto/visualize/profiles.py]: Ah, totally got your comment wrong, now I see!
Well, okay, that'll be done in 5 minutes

**Pull request merged**